File: src/main/java/my/study/repository/design/pattern/adapter/UseDemo.java
package my.study.repository.design.pattern.adapter;

/**
 * @author djh on  2018/8/10 00:05
 * @E-Mail 1544579459@qq.com
 */
public class UseDemo {
    private static class Mobile {
        private void charge(int voltage) {
            if (voltage == 5) {
                System.out.println("电压匹配成功!");
            } else {
                System.out.println("高电压");
            }
        }
    }

    public static void main(String[] args) {
        ObjectAdapter objectAdapter = new ObjectAdapter(new Src());
        Mobile mobile = new Mobile();
        mobile.charge(objectAdapter.mobileCharge());
    }
}

File: src/main/java/my/study/repository/design/pattern/adapter/kotlin/UseDemo.kt
package my.study.repository.design.pattern.adapter.kotlin

/**
 * @author djh on  2018/10/4 16:16
 * @E-Mail 1544579459@qq.com
 */
object UseDemo {

    private class Mobile {
        fun startCharge(voltage: Int) {
            if (voltage == 55) {
                println("电压适配成功,开始充电!")
            } else {
                println("$voltage V 电压太高!")
            }
        }
    }

    @JvmStatic
    fun main(args: Array<String>) {
        val objectAdapter = ObjectAdapter(Src())
        val mobile = Mobile()
        mobile.startCharge(objectAdapter.mobileCharge())
    }
}